### PR TITLE
Fusetools 2307 graphical remote debug for jmx routes basic

### PR DESCRIPTION
**Pull request by apupier**
it allows to put breakpoint on a remote camel context.
They will be triggered if a Remote Camel Debug laucnh configuration has been used specifiying the correct - almost hidden - files.
So not perfect but it is now possible.
More to come in another PR with https://issues.jboss.org/browse/FUSETOOLS-2317

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/behaviours/CamelPersistencyBehaviour.java [hunk 1]
```diff
@@ -88,15 +88,10 @@ public void saveDiagram(IProgressMonitor monitor) {
 	 * Creates the diagram from the current model
 	 */
 	public boolean loadModel(IEditorInput input) {
-		IFile camelContextFile = null;
-		
 		IFileEditorInput fileEditorInput = editor.asFileEditorInput(input);
 		if (fileEditorInput != null) {
-			camelContextFile = fileEditorInput.getFile();
-			if (camelContextFile.getFullPath().toOSString().indexOf(".FuseRemoteCamelContextData") == -1) {
```
Comment (jpav) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/behaviours/CamelPersistencyBehaviour.java]: Disclaimer: I don't know enough about the history of this code to know whether it's okay to remove this.

File: core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/io/CamelContextNodeEditorInput.java [hunk 2]
```diff
@@ -89,4 +89,5 @@ public void pushbackToRemoteContext(String xml) {
 	private boolean isCamelContextNode(Node contextNode) {
 		return "org.fusesource.ide.jmx.camel.navigator.CamelContextNode".equals(contextNode.getClass().getName());
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/io/CamelContextNodeEditorInput.java]: shouldn't we make a constant out of this string?

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelDesignEditor.java [hunk 3]
```diff
@@ -541,7 +540,9 @@ public void refreshDiagramContents(Diagram diagram) {
 	public void modelChanged() {
 		// we only update if the correct editor tab is selected
 		parent.setDirtyFlag(true);
-		if (getParent().getActivePage() != CamelEditor.DESIGN_PAGE_INDEX) return;
+		if (getParent().getActivePage() != CamelEditor.DESIGN_PAGE_INDEX){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelDesignEditor.java]: don't want to be picky but I prefer a whitespace between ) and {

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java [hunk 4]
```diff
@@ -255,7 +257,9 @@ public void documentAboutToBeChanged(DocumentEvent event) {
 	 */
 	@Override
 	public void documentChanged(DocumentEvent event) {
-		if (getActivePage() == SOURCE_PAGE_INDEX) 	setDirtyFlag(true);
+		if (getActivePage() == SOURCE_PAGE_INDEX){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java]: white space between ) and {

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java [hunk 5]
```diff
@@ -455,8 +459,10 @@ public void dispose() {
 	@Override
 	protected void pageChange(int newPageIndex) {
 		if (newPageIndex == SOURCE_PAGE_INDEX) {
-			if (sourceEditor == null) sourceEditor = new StructuredTextEditor();
-			if (rollBackActive == false) {
+			if (sourceEditor == null){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java]: whitespace

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java [hunk 6]
```diff
@@ -479,8 +485,9 @@ protected void pageChange(int newPageIndex) {
 					} else {
 						updateModelFromSource();
 						lastError = "";
-						if (newPageIndex == GLOBAL_CONF_INDEX) globalConfigEditor.reload();
-						if (newPageIndex == DESIGN_PAGE_INDEX) {
+						if (newPageIndex == GLOBAL_CONF_INDEX){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java]: whitespace

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java [hunk 7]
```diff
@@ -742,7 +748,9 @@ protected void handlePropertyChange(int propertyId) {
 	@Override
 	protected void setInput(IEditorInput input) {
 		super.setInput(input);
-		if (input instanceof CamelXMLEditorInput && input.equals(this.editorInput) == false) this.editorInput = (CamelXMLEditorInput)input;
+		if (input instanceof CamelXMLEditorInput && !input.equals(this.editorInput)){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelEditor.java]: whitespace

File: core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/io/CamelContextNodeEditorInput.java (hunk 2 above)
Comment (apupier) [inline core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/io/CamelContextNodeEditorInput.java]: I made even better in my other PR, I moved the class and this equal is not even needed so i removed it

File: editor/plugins/org.fusesource.ide.launcher/src/org/fusesource/ide/launcher/debug/util/CamelDebugUtils.java [hunk 8]
```diff
@@ -193,7 +193,9 @@ public static String getCondition(IBreakpoint breakpoint) {
 	 * @return		the project or null if not able to determine a project
 	 */
 	public static IProject getProjectForFilePath(String filePath) {
-		if (Strings.isBlank(filePath)) return null;
+		if (Strings.isBlank(filePath)){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.launcher/src/org/fusesource/ide/launcher/debug/util/CamelDebugUtils.java]: whitespace

File: editor/plugins/org.fusesource.ide.launcher/src/org/fusesource/ide/launcher/debug/util/CamelDebugUtils.java [hunk 9]
```diff
@@ -212,7 +214,9 @@ public static String getRawCamelContextFilePathFromLaunchConfig(ILaunchConfigura
 		if (launchConfig != null) {
 			try {
 				String fileUnderDebug = launchConfig.getAttribute(CamelContextLaunchConfigConstants.ATTR_FILE, (String)null);
-				if (fileUnderDebug.startsWith("file:")) fileUnderDebug = fileUnderDebug.substring("file:".length());
+				if (fileUnderDebug.startsWith("file:")){
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.launcher/src/org/fusesource/ide/launcher/debug/util/CamelDebugUtils.java]: whitespace

File: jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java [hunk 10]
```diff
@@ -250,6 +250,10 @@ public void editRoutes() {
 				} catch (PartInitException e) {
 					CamelJMXPlugin.getLogger().warning("Could not open editor: " + CAMEL_EDITOR_ID + ". Reason: " + e, e);
 				}
+				
+				//RemoteCamel
```
Comment (lhein) [inline jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java]: code smell?

File: jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java [hunk 11]
```diff
@@ -578,5 +582,5 @@ public int hashCode() {
 		}
 		return super.hashCode();
 	}
-
+
```
Comment (lhein) [inline jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java]: ???

File: jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java (hunk 10 above)
Comment (apupier) [inline jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java]: effectively, it is fixed in next PR

File: jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java (hunk 11 above)
Comment (apupier) [inline jmx/plugins/org.fusesource.ide.jmx.camel/src/org/fusesource/ide/jmx/camel/navigator/CamelContextNode.java]: already fixed in next PR

**Comment by apupier** [comment]
I added a space in reported place in https://github.com/fusesource/fuseide/pull/914

**Pull request merged**